src/StackExchange.Redis/RedisServer.cs

```diff
@@ -575,7 +575,7 @@ internal override Task<T> ExecuteAsync<T>(Message message, ResultProcessor<T> pr
         {   // inject our expected server automatically
             if (server == null) server = this.server;
             FixFlags(message, server);
-            if (!server.IsConnected)
+            if (!server.IsConnected && !multiplexer.RawConfig.BacklogPolicy.QueueWhileDisconnected)
```

Comment:
This particular change felt surprising to me. FireAndForget would only be FireAndForget if we're connected? Sort of makes sense, but.... just from the most cursory read of this logic, feels like there is a risk that it could lead to _blocking_ behavior which might be a surprise to people using fire and forget because they want nonblocking behavior?

Reply:
We need to do _something_ here to not throw but I agree those 2 cases aren't quite right - good catch! I think this should be adjusted slightly in that we only want to wrap in the "throw to bail" in both this and ExecuteSync below - will adjust/test and ping for re-review here shortly :)

Reply:
@TimLovellSmith alrighty I think this is as it should be now, can you please give eyes/thoughts on it again when time allows?